common/component.go

```diff
@@ -230,7 +250,8 @@ func (c *VcapComponent) ListenAndServe() {
 	}
 
 	f := func(user, password string) bool {
-		return user == c.Varz.Credentials[0] && password == c.Varz.Credentials[1]
+		return (user == c.Varz.Credentials[0] && password == c.Varz.Credentials[1]) ||
```

Comment:
Update to ensure we are only authing if using the correct credentials for the correct endpoint. A correct broker auth would get you access to /routes even if a separate StatusConfig was set.

Reply:
The block `if c.Varz.Credentials == nil`will also need to get removed.

Reply:
That block has been terminated and the auth function has been modified.